### PR TITLE
Update RootProviderRegistry factory methods

### PR DESCRIPTION
**Pull request by tgroh**
Rename `#default` to `#javaNative`

Add an `impulseRegistry` that handles only impulse, for the portable runner.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by tgroh** [comment]
See #5123 for reasoning

R: @huygaa11

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/RootProviderRegistry.java [hunk 1]
```diff
@@ -28,25 +28,38 @@
...
+                ReadEvaluatorFactory.inputProvider(context, options))
+            .put(DIRECT_TEST_STREAM_URN, new TestStreamEvaluatorFactory.InputProvider(context))
+            .put(FLATTEN_TRANSFORM_URN, new EmptyInputProvider())
+            .build());
+  }
+
+  /**
+   * Returns a {@link RootProviderRegistry} that only supports the {@link Impulse} primitive.
+   */
+  public static RootProviderRegistry impulseRegistry(EvaluationContext ctxt) {
```
Comment (huygaa11) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/RootProviderRegistry.java]: This file has inconsistent uses of "ctxt" vs. "context". I would stick to one of them.

Comment (tgroh) [reply runners/direct-java/src/main/java/org/apache/beam/runners/direct/RootProviderRegistry.java]: Done.

**Comment by tgroh** [comment]
run java precommit

**Pull request merged**